src/main/java/org/example/learningsystem/core/audit/config/AuditorAwareImpl.java

```diff
@@ -0,0 +1,57 @@
+package org.example.learningsystem.core.audit.config;
+
+import org.springframework.data.domain.AuditorAware;
+import org.springframework.security.authentication.AnonymousAuthenticationToken;
+import org.springframework.security.core.Authentication;
+import org.springframework.security.core.context.SecurityContextHolder;
+import org.springframework.security.core.userdetails.User;
+import org.springframework.security.oauth2.server.resource.authentication.JwtAuthenticationToken;
+
+import java.util.Optional;
+
+import static java.util.Objects.nonNull;
+import static org.springframework.security.oauth2.core.OAuth2TokenIntrospectionClaimNames.CLIENT_ID;
+
+public class AuditorAwareImpl implements AuditorAware<String> {
+
+    private static final String USERNAME_CLAIM = "user_name";
+
+    @Override
+    public Optional<String> getCurrentAuditor() {
+        var context = SecurityContextHolder.getContext();
+
+        var authentication = context.getAuthentication();
+        if (!isAuthenticated(authentication)) {
+            return Optional.empty();
+        }
```

Comment:
there is no need to check authentication, as this code will not be reached in case of authentication failure in security filter